src/pages/CoachProfile/BioEditor/BioEditor.tsx

```diff
@@ -0,0 +1,124 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import React, { useEffect, useState } from 'react';
+import { ErrorPage } from '../../ErrorPage/ErrorPage';
+import { Spinner } from '../../../components/Spinner/Spinner';
+import axios, { AxiosError } from 'axios';
+import { getCookie } from 'typescript-cookie';
+import { Button } from '../../../components/Button';
+import { Cancel, Check, Settings } from '@mui/icons-material';
+import { ButtonBoxStyles, ButtonIconStyles, CardStyles } from './BioEditor.styles';
+
+type BioEditorProps = {
+  showSnackbar: (message: string, severity: 'success' | 'error') => void;
+};
+
+export const BioEditor = ({ showSnackbar }: BioEditorProps) => {
+  const [editMode, setEditMode] = useState(false);
+  const [updatedDescription, setDescription] = useState('');
+  const queryClient = useQueryClient();
+
+  const toggleEditMode = () => {
+    setEditMode(!editMode);
+  };
+
+  const handleUpdate = (description: string) => {
+    mutation.mutate(description);
+    toggleEditMode();
+  };
+
+  const { data, isLoading, isFetched, isError, error } = useQuery(
+    ['coach-description'],
+    async () => {
+      const { data } = await axios.get('http://localhost:8081/api/user/coach/description', {
+        headers: { Authorization: `Bearer ${getCookie('userToken')}` },
+      });
+      return data;
+    }
+  );
+
+  useEffect(() => {
+    if (isFetched && data) {
+      setDescription(data); // Ustawienie pobranej wartości jako domyślna wartość w polu TextArea
+    }
+  }, [isFetched, data]);
+
+  /**
+   * sends update to backend and refetches updated data
+   */
+  const mutation = useMutation(
+    (description: string) => {
+      return axios.post(
+        'http://localhost:8081/api/user/coach/change/description',
+        { description: description },
+        {
+          headers: { Authorization: `Bearer ${getCookie('userToken')}` },
+        }
+      );
+    },
+    {
+      onSuccess: () => {
+        showSnackbar('Your description has been updated!', 'success');
+      },
+      onError: error => {
+        if (error instanceof AxiosError) {
+          showSnackbar(error.message, 'error');
+        }
+      },
+      onSettled: () => {
+        queryClient.invalidateQueries(['coach-description']);
+      },
+    }
+  );
+
+  if (isError && error instanceof Error) {
+    return <ErrorPage message={error.message} />;
+  }
+
+  if (isLoading) {
+    return <Spinner message="Loading profile" />;
+  }
+  return (
+    <Card sx={CardStyles} elevation={3}>
+      <CardContent>
+        <Typography variant="h5" color="primary.main">
+          Your description:
+        </Typography>
+        <Typography variant="body2" sx={{ pb: 3 }}>
+          This is the message all users will see upon loading your profile
+        </Typography>
+        {isFetched && data && !editMode ? (
+          <Typography>{data}</Typography>
+        ) : isFetched && data && editMode ? (
```

Comment:
Czy jest może inny sposób, w który moglibyśmy obskoczyć tą logikę? ternary jest trochę nieczytelne tutaj

Reply:
Ciężko mi coś wymyślić poza zamienieniem `isFetched && data && !editMode` na jakąś zmienną. Jak masz jakiś pomysł to zapodaj i spróbuję zrobić